go/dpagg/sum_test.go

```diff
@@ -903,6 +903,207 @@ func TestThresholdedResultFloat64(t *testing.T) {
 	}
 }
 
+// Tests ComputeConfidenceInterval that If both bounds are non-negative, then trim any negative interval. Similarly, if both
+// bounds are non-positive then trim any positive interval from the confidence interval.
+func TestSumComputeConfidenceIntervalForInt64PostProcessing(t *testing.T) {
+	// noNoise is set to skip initial argument checking.
+	noNoise := noNoise{}
+	for _, tc := range []struct {
+		opt     *BoundedSumInt64Options
+		confInt noise.ConfidenceInterval // Raw confidence interval.
+		want    noise.ConfidenceInterval // Post-processing.
+	}{
+		{
+			opt:     &BoundedSumInt64Options{Lower: 2, Upper: 3, Noise: noNoise},
+			confInt: noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+			want:    noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+		},
+		{
+			opt:     &BoundedSumInt64Options{Lower: 2, Upper: 3, Noise: noNoise},
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: 5},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: 5},
+		},
+		{
+			opt:     &BoundedSumInt64Options{Lower: -3, Upper: -2, Noise: noNoise},
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: 5},
+			want:    noise.ConfidenceInterval{LowerBound: -5, UpperBound: 0},
+		},
+		{
+			opt:     &BoundedSumInt64Options{Lower: -2, Upper: 3, Noise: noNoise},
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: 5},
+			want:    noise.ConfidenceInterval{LowerBound: -5, UpperBound: 5},
+		},
+		// Infinite bounds happens for extremely small alpha for Gaussian.
+		{
+			opt:     &BoundedSumInt64Options{Lower: 2, Upper: 3, Noise: noNoise},
+			confInt: noise.ConfidenceInterval{LowerBound: math.Inf(-1), UpperBound: math.Inf(1)},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: math.Inf(1)},
+		},
+	} {
+		c := NewBoundedSumInt64(tc.opt)
+		// This makes Noise interface return the raw confidence interval when ComputeConfidenceIntervalInt64 is called.
+		c.noise = getMockConfInt(tc.confInt)
+
+		c.Result()
+		got, _ := c.ComputeConfidenceInterval(0.1) // alpha is ignored in mockConfInt.
+
+		if got.LowerBound != tc.want.LowerBound {
+			t.Errorf("TestSumComputeConfidenceIntervalForInt64PostProcessing(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, LowerBounds are not equal",
+				tc.confInt.LowerBound, tc.confInt.UpperBound, got.LowerBound, tc.want.LowerBound)
+		}
+		if got.UpperBound != tc.want.UpperBound {
+			t.Errorf("TestSumComputeConfidenceIntervalForInt64PostProcessing(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, UpperBounds are not equal",
+				tc.confInt.LowerBound, tc.confInt.UpperBound, got.UpperBound, tc.want.UpperBound)
+		}
+	}
+}
+
+// Tests ComputeConfidenceInterval that If both bounds are non-negative, then trim any negative interval. Similarly, if both
```

Comment:
nit: `Tests that ComputeConfidenceInterval for float64 trims negative intervals if both bounds are non-negative; and trims positive intervals if both bounds are non-positive.`